ml/src/main/spark_2.1/org/apache/spark/ml/h2o/SparkSpecificUtils.scala

```diff
@@ -0,0 +1,57 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.apache.spark.ml.h2o
+
+import org.apache.spark.SparkContext
+import org.apache.spark.ml.h2o.models.CrossSparkUtils
+import org.apache.spark.ml.param.Params
+import org.apache.spark.ml.util.DefaultParamsReader
+import org.apache.spark.ml.util.DefaultParamsReader.Metadata
+import org.json4s.jackson.JsonMethods._
+import org.json4s.{DefaultFormats, JObject}
+
+object SparkSpecificUtils extends CrossSparkUtils {
+
+  private def getAndSetParams(
+                               instance: Params,
+                               metadata: Metadata,
+                               skipParams: Option[List[String]] = None): Unit = {
+    implicit val format = DefaultFormats
```

Comment:
Can't we use this piece of code for 2.3 and 2.4 as well?

Reply:
I was thinking about that, but that would complicate source directory definitions. We would need to have multiple CrossSpark traits and source directories named like spark-2.1-2.3. Doable, but not sure it it brings any real benefit compared to complexity it brings

Reply:
If you have any idea how to make this better, please let me know :)

Reply:
Maybe we should leave it as it's for now. I don't know the whole context.

Reply:
I will still think about it, I don't like code duplication, but was having hard time creating nice solution. Maybe we can brainstorm a bit tomorrow? :)

Reply:
Thank for today's discussion -> was able to simplify even further